torch/nn/init.py

```diff
@@ -273,85 +435,140 @@ def kaiming_normal(tensor, a=0, mode='fan_in'):
 
     Examples:
         >>> w = torch.Tensor(3, 5)
-        >>> nn.init.kaiming_normal(w, mode='fan_out')
+        >>> nn.init.kaiming_normal_(w, mode='fan_out')
+        >>> he_init = nn.init.kaiming_normal_(mode='fan_out')
+        >>> he_init(w)
     """
-    fan = _calculate_correct_fan(tensor, mode)
-    gain = calculate_gain('leaky_relu', a)
-    std = gain / math.sqrt(fan)
-    with torch.no_grad():
-        return tensor.normal_(0, std)
+    def __init__(self, a=0, mode="fan_in"):
+        gain = calculate_gain('leaky_relu', a) ** 2
+        self.vs = VarianceScaling(
```

Comment:
It seems that you could make `VarianceScaling` an `Initializer` subclass, and then subclass from it in here. Then, in `__init__` you'd take those arguments, modify them, and pass them to `super`. Then, you don't even need to implement `__call__`